R/gghist.R

```diff
@@ -6,11 +6,78 @@
 #' @param variable The column name inside data that will be made into a histogram.
 #'
 #' @return Produces a ggplot histogram with mean, median and standard deviation indicated on the plot.
+#'
 #' @export
+#' @importFrom ggplot2 geom_histogram
+#' @importFrom ggplot2 geom_vline
+#' @importFrom ggplot2 annotate
 #'
 #' @examples
 #' gghist(iris, Sepal.Length)
 gghist <- function(data, variable) {
-  # guts of function goes here
+  #todo: raise an exception about the data type of input variable. must be continuous 1D vector
+  #todo: process the variable column in some way to find a good way to set the binwidth in geom_hist???
+  #todo:allow the user to specify the number of bins; need a bins argument with a default
+
+  # check that data is correct datatype
+  if (!class({{data}})[1] %in% c("tbl_df", "tbl", "data.frame")) {
```

Comment:
We need to write a test for each branch. So in this case a test should be written. Do if you have time!